includes/senders/BaseSender.php

```diff
@@ -27,10 +27,16 @@ public function __construct( $name, $image = null ) {
 	 * @return array
 	 */
 	public function jsonSerialize() {
-		return array(
-			'name'  => $this->name,
-			'image' => $this->image,
+
+		$data = array(
+			'name' => $this->name,
 		);
+
+		if ( $this->image ) {
+			$data[ get_class( $this->image ) ] = $this->image;
+		}
+
+		return array_filter( $data );
```

Comment:
I am not sure what the `array_filter` function is supposed to do without passing it a callback function. Would you explain its usage here?